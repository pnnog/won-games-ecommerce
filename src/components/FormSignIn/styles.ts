import styled, { css } from 'styled-components';

export const ForgotLink = styled.a`
  ${({ theme }) => css`
    align-self: end;
    cursor: pointer;
    margin-bottom: calc(${theme.spacings.medium}*2);

    font-size: ${theme.font.sizes.xsmall};
    text-decoration: none;
    color: ${theme.colors.black};
  `}
`;
