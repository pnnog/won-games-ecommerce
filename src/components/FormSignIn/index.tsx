import TextField from 'components/TextField';
import * as S from './styles';
import { Email, Lock } from '@styled-icons/material-outlined';
import Button from 'components/Button';
import Link from 'next/link';
import { FormWrapper, FormLink } from 'components/Form';

const FormSignIn = () => (
  <FormWrapper>
    <form>
      <TextField
        type="email"
        name="email"
        placeholder="Email"
        icon={<Email />}
      />
      <TextField
        type="password"
        name="password"
        placeholder="Password"
        icon={<Lock />}
      />
      <S.ForgotLink href="#">Forgot your password?</S.ForgotLink>

      <Button size="large" fullWidth>
        Sign in now
      </Button>

      <FormLink>
        Donâ€™t have an account? <Link href={'/sign-up'}>Sign up</Link>
      </FormLink>
    </form>
  </FormWrapper>
);

export default FormSignIn;
