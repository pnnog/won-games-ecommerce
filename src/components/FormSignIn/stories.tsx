import { Meta, Story } from '@storybook/react';
import FormSignIn from '.';

export default {
  title: 'Forms/FormSignIn',
  component: FormSignIn,
  args: {},

  parameters: {
    viewport: {
      defaultViewport: 'mobile1',
    },

    layout: {
      default: 'fullscreen',
    },
  },
} as Meta;

export const Default: Story = (args) => (
  <div style={{ width: '30rem', margin: '2rem auto' }}>
    <FormSignIn {...args} />
  </div>
);
