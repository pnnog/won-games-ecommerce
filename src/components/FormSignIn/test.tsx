import { screen } from '@testing-library/react';

import FormSignIn from '.';
import { renderWithTheme } from 'utils/tests/helpers';

//mockar componentes externos e testar
//testar links

jest.mock('components/TextField', () => ({
  __esModule: true,
  default: () => <div data-testid="text field mock"></div>,
}));

jest.mock('components/Button', () => ({
  __esModule: true,
  default: () => <div data-testid="button mock"></div>,
}));

describe('<FormSignIn />', () => {
  it('should render FormSignIn', () => {
    const { container } = renderWithTheme(<FormSignIn />);

    expect(screen.getAllByTestId('text field mock')).toHaveLength(2);
    expect(screen.getByTestId('button mock')).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /forgot your password?/i })
    ).toHaveAttribute('href', '#');

    expect(screen.getByRole('link', { name: /Sign up/ })).toHaveAttribute(
      'href',
      '/sign-up'
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
