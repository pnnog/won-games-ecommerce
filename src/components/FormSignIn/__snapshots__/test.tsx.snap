// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignIn /> should render FormSignIn 1`] = ` `;
